native/src/boot/bootimg.cpp

```diff
@@ -617,20 +664,32 @@ void repack(const char *src_img, const char *out_img, bool skip_comp) {
     file_align();
 
     // ramdisk
-    off.ramdisk = lseek(fd, 0, SEEK_CUR);
-    if (boot.flags[MTK_RAMDISK]) {
-        // Copy MTK headers
-        xwrite(fd, boot.r_hdr, sizeof(mtk_hdr));
-    }
-    if (access(RAMDISK_FILE, R_OK) == 0) {
-        auto m = mmap_data(RAMDISK_FILE);
-        if (!skip_comp && !COMPRESSED_ANY(check_fmt(m.buf, m.sz)) && COMPRESSED(boot.r_fmt)) {
-            hdr->ramdisk_size() = compress(boot.r_fmt, fd, m.buf, m.sz);
-        } else {
-            hdr->ramdisk_size() = xwrite(fd, m.buf, m.sz);
+    for(int i = 0; i < boot.num_vendor_ramdisk_table_entries; ++i) {
+        off.ramdisk = lseek(fd, 0, SEEK_CUR);
+        if (boot.flags[MTK_RAMDISK]) {
+            // Copy MTK headers
+            xwrite(fd, boot.r_hdr, sizeof(mtk_hdr));
+        }
+        std::string file_name(RAMDISK_FILE);
+        if (boot.hdr->header_version() == 4 && boot.hdr->is_vendor){
+            file_name.assign("ramdisk.");
+            file_name.append(to_string(i));
+            file_name.append(".cpio");
+        }
```

Comment:
```suggestion
          std::string file_name(RAMDISK_FILE);
          char file_name[PATH_MAX] = {};
          ssprintf(file_name, sizeof(file_name), VENDOR_RAMDISK_FILE, i);
```

Reply:
Switched to ssprintf, but change file name only for vendor_boot v4 to keep backwards compatibility with scripts etc.